sway-ir/tests/tests.rs

```diff
@@ -26,14 +26,14 @@ fn ir_to_ir_tests() {
                 } else {
                     panic!(
                         "File which doesn't match valid passes: {:?}",
-                        path.file_name().unwrap_or_else(|| path.as_os_str())
+                        path.file_name().unwrap_or(path.as_os_str())
                     );
                 }
             }
             Some("out_ir") => (),
             _ => panic!(
                 "File with invalid extension in tests dir: {:?}",
-                path.file_name().unwrap_or_else(|| path.as_os_str())
+                path.file_name().unwrap_or(path.as_os_str())
```

Comment:
We ended up landing these clippy nits in #874 as a few of us were running into them! Could you remove these and maybe rebase onto master?

Reply:
I think we don't rebase at Fuel in general, at least John advised against it (breaks the reviews). But sure I'm going to merge master in.